GameEngine/Entities/Animal.cs

```diff
@@ -1,220 +1,61 @@
-        public string? Letter { get; set; }
-
-        private static int id = 0;
-
-        /// <summary>
-        /// Generates new id.
-        /// </summary>
-        /// <returns>Incremented id.</returns>
-        protected int GenerateId() => id++;
-
-        /// <summary>
-        /// Moves current animal on the board.
-        /// </summary>
-        /// <param name="board">Board.</param>
-        /// <param name="animal">Animal.</param>
-        /// <param name="animals">Animals.</param>
-        public abstract void MoveAnimal(Board board, Animal animal, List<Animal> animals);
-
-        /// <summary>
-        /// Gives birth of a new animal.
-        /// </summary>
-        /// <param name="board">Board.</param>
-        /// <param name="animals">Animals.</param>
-        /// <returns>New animal (child).</returns>
-        public abstract Animal? GiveBirth(Board board, List<Animal> animals);
-
-        /// <summary>
-        /// Makes next move of animal.
-        /// </summary>
-        /// <param name="animal">Animal.</param>
-        /// <param name="freeCellsToMove">Free cells to move.</param>
-        protected void MakeNextMove(Animal animal, List<NewAnimalCoordinates> freeCellsToMove)
-        {
-            MoveToNewRandomPosition(animal, freeCellsToMove);
-            Die(animal, DecreaseHealth(animal));
-        }
-
-        /// <summary>
-        /// Moves animal to a new random position.
-        /// </summary>
-        /// <param name="animal">Animal.</param>
-        /// <param name="freeCellsToMove">Free cells to move.</param>
-        protected void MoveToNewRandomPosition(Animal animal, List<NewAnimalCoordinates> freeCellsToMove)
-        {
-            NewAnimalCoordinates nextCoordinatesToMove = freeCellsToMove[Helper.random.Next(0, freeCellsToMove.Count)];
-            animal.CoordinateX = nextCoordinatesToMove.NewXCoordinate;
-            animal.CoordinateY = nextCoordinatesToMove.NewYCoordinate;
-        }
 
         /// <summary>
-        /// Decreases animal's health.
+        /// Heals the animal for maximum health.
         /// </summary>
-        /// <param name="currentAnimal">Current animal.</param>
-        /// <returns>Decreased health.</returns>
-        private double DecreaseHealth(Animal currentAnimal) => currentAnimal.Health -= ConstantsRepository.HealthDecreaser;
+        protected virtual void Heal() => Health = ConstantsRepository.MaxHealth;
 
         /// <summary>
-        /// Makes the animal dead.
+        /// Heals the animal for the passed value.
         /// </summary>
-        /// <param name="currentAnimal">Current animal.</param>
-        /// <param name="currentHealth">Current health.</param>
-        private void Die(Animal currentAnimal, double currentHealth)
-        {
-            if (currentHealth <= 0)
-            {
-                currentAnimal.IsDead = true;
-            }
-        }
+        /// <param name="healthToIncrease"></param>
+        /// <returns></returns>
```

Comment:
Either specify what is returned or remove tags as they are redundant.

Reply:
Done.